src/gui/qgsgeometryrubberband.h

```diff
@@ -86,12 +92,12 @@ class GUI_EXPORT QgsGeometryRubberBand: public QgsMapCanvasItem
     };
 
     QgsGeometryRubberBand( QgsMapCanvas *mapCanvas, QgsWkbTypes::GeometryType geomType = QgsWkbTypes::LineGeometry );
-    ~QgsGeometryRubberBand() override;
+    virtual ~QgsGeometryRubberBand() override;
 
     //! Sets geometry (takes ownership). Geometry is expected to be in map coordinates
-    void setGeometry( QgsAbstractGeometry *geom SIP_TRANSFER );
+    virtual void setGeometry( QgsAbstractGeometry *geom SIP_TRANSFER );
```

Comment:
Why is this virtual? There's no derived classes of QgsGeometryRubberBand to use it...

Reply:
the new class QgsMapToolCaptureRubberband, derived from QgsGeometryRubberBand, override to make it private. The aim is to not exposed the setting of the geometry in the derived class.